### ISSUE TITLE
Incorrect type inference for super keyword

### ISSUE THREAD

**Issue by ENvironmentSet** [microsoft/TypeScript#33387]
**TypeScript Version:**  3.6.2

**Search Terms:**
type inference for super keyword
super keyword type inference
**Code**

```ts
class X {
  constructor(public x: string) {}
}

class Y extends X {
  toString(): string {
    return super.x;
  }
}
```

**Expected behavior:**

#### Summary

I expected that compiler throws type error that says `super.x` is not assignable to type `string`.

#### Long explanation

According to ECMAScript 10 specification, [[HomeObject]] -- internal slot that contains the object where this function is defined, which used to get prototype of itself, where super property lookups begin -- of `Y.prototype.toString`(or method `toString`) is `Y.prototype`. and prototype of `Y.prototype` is `X.prototype`. As `x` doesn't exist on `X.prototype`, `super.x` must not be inferred and compiler must throw an error about this.

**Actual behavior:**

this code passes type checking well.
Also, compiler infers type of `super` as `X`, which can't be type of `super`.

**Playground Link:** 
[typescript playground](http://www.typescriptlang.org/play/?target=5#code/MYGwhgzhAEAa0G8BQ1rAPYDsIBcBOArsDungBQAOBARiAJbDQAeAXNLnnZgOYCUiAXyRCkoSDACa0AKZMc0zABMY8ZKhIBlfF25lebDjsQpU0PNJwE8mdgQrS8AOiYBuE0KFA)
[flow playground](https://flow.org/try/?fbclid=IwAR1t05_Tq01_KRB4ccNngL-LLdEJCdxHRV-ZZDgg6XyFn99jcRnmdRjQtKk#0PQKgBAAgZgNg9gdzCYAoVBjGBDAzrsADTAG9UwwAPALjFwBcAnASwDsBzcsDOVhxgK4Z6cRgAoadJm3YBKUlwr0AFs1wA6SmAC8VANxcAvqmOYc+MAE0wAU0r0brACYFiZJXADK0jmPnuKMEYbegFGVjoBAAcbRk0DCmNTHj44GBt1eHYxVhskSzEAImwAIwxC2XURbxZfWVkgA)(FYI, Same thing happens in flow either.)

**Comment by Ailrun** [comment]
Duplicate of https://github.com/microsoft/TypeScript/issues/30516 (and of other issues linked in the issue)
